### PR TITLE
Homework : hwdetect

### PR DESCRIPTION
**Pull request by CheeseMite**
Script for detecting new devices and printing some info.  

`> DEVICE_NAME ` means that particular device has just connected.
`< DEVICE_NAME ` means that particular device has just disconnected.

If the description of device is available, script prints it just after the name of device.
 
### Example 
```
> sdb
Block device
/0/100/8.1/0.3/0/3/0.0.0      /dev/sdb   disk           16GB Silicon-Power16G
> sg1
< sdb
Block device
< sdb1
< sg1
> ttyUSB0
Usb to ttl convertor
< serial
< ttyUSB0
Usb to ttl convertor
```

### CODE REVIEW

File: hwdetect/hwdetect.sh [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+		return $USB_CVT
+	fi
+
+	if [[ "$dev" == *"sd"? ]]
+	then
+		return $BLK_DEV
+	fi
+
+	return $OTHER
+}
```
Comment (yekovalyov) [inline hwdetect/hwdetect.sh:24]: Is it possible to exclude this function and use just device name without enumeration?

File: hwdetect/hwdetect.sh [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    	then
+	    	sleep .5
+            printInfo "$difs"
+		    devs=`ls /dev/`
+	    fi	
+    done
+    return $STATUS_OK
+}
+
+main
```
Comment (yekovalyov) [inline hwdetect/hwdetect.sh]: Some old tools required last empty line in program sources.

Comment (CheeseMite) [reply hwdetect/hwdetect.sh]: Thanks, fixed

File: hwdetect/hwdetect.sh (hunk 1 above)
Comment (CheeseMite) [reply hwdetect/hwdetect.sh:24]: Yes, then `printInfo` will look like this :
```
function printInfo
{
	IFS=$'\n'
	cur=${1}
	
	for dev in $cur
	do
		echo $dev

		case $dev in
			*ttyUSB*)
			       	echo "Usb to ttl convertor"
				;;
			*sda?)
			       	echo "Block device"
					sudo lshw -short | grep ${dev//[<> ]} -m 1
				;;
			*)
			       	continue
				;;
		esac

	done
}
```

In this case it's better, I think. But in a case, where are 2+ section depended on device type trick with a enumeration looks better. 
And probably there are cases, where only name of device is not enough and having function with device type identification is must-have.

Should I update the code?

**Pull request closed without merging**